pkg/syncer/apiimporter.go

```diff
@@ -134,44 +134,71 @@ type APIImporter struct {
 func (i *APIImporter) Start(ctx context.Context, pollInterval time.Duration) {
 	defer runtime.HandleCrash()
 
-	i.kcpInformerFactory.Start(ctx.Done())
-	i.kcpInformerFactory.WaitForCacheSync(ctx.Done())
+	logger := logging.WithReconciler(klog.FromContext(ctx), "kcp-syncer-apiimporter")
+	ctx = klog.NewContext(ctx, logger)
 
-	klog.Infof("Starting API Importer for location %s in cluster %s", i.location, i.logicalClusterName)
+	logger.Info("Starting API Importer", "location", i.location, "cluster", i.logicalClusterName)
 
 	clusterContext := request.WithCluster(ctx, request.Cluster{Name: i.logicalClusterName})
 	go wait.UntilWithContext(clusterContext, func(innerCtx context.Context) {
 		i.ImportAPIs(innerCtx)
 	}, pollInterval)
 
 	<-ctx.Done()
-	i.Stop()
+	i.Stop(ctx)
 }
 
-func (i *APIImporter) Stop() {
-	klog.Infof("Stopping API Importer for location %s in cluster %s", i.location, i.logicalClusterName)
+func (i *APIImporter) Stop(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+	logger.Info("Stopping API Importer", "location", i.location, "cluster", i.logicalClusterName)
 
 	objs, err := i.apiresourceImportIndexer.ByIndex(
 		clusterctl.LocationInLogicalClusterIndexName,
 		clusterctl.GetLocationInLogicalClusterIndexKey(i.location, i.logicalClusterName),
 	)
 	if err != nil {
-		klog.Errorf("error trying to list APIResourceImport objects for location %s in logical cluster %s: %v", i.location, i.logicalClusterName, err)
+		logger.Error(err, "error trying to list APIResourceImport objects", "location", i.location, "cluster", i.logicalClusterName)
 	}
 	for _, obj := range objs {
 		apiResourceImportToDelete := obj.(*apiresourcev1alpha1.APIResourceImport)
 		err := i.kcpClusterClient.Cluster(i.logicalClusterName).ApiresourceV1alpha1().APIResourceImports().Delete(request.WithCluster(context.Background(), request.Cluster{Name: i.logicalClusterName}), apiResourceImportToDelete.Name, metav1.DeleteOptions{})
 		if err != nil {
-			klog.Errorf("error deleting APIResourceImport %s: %v", apiResourceImportToDelete.Name, err)
+			logger.Error(err, "error deleting APIResourceImport", "apiResourceImport", apiResourceImportToDelete.Name)
 		}
 	}
 }
 
 func (i *APIImporter) ImportAPIs(ctx context.Context) {
-	klog.Infof("Importing APIs from location %s in logical cluster %s (resources=%v)", i.location, i.logicalClusterName, i.resourcesToSync)
-	crds, err := i.schemaPuller.PullCRDs(ctx, i.resourcesToSync...)
+	logger := klog.FromContext(ctx)
+	syncTargetKey, err := kcpcache.MetaClusterNamespaceKeyFunc(&metav1.PartialObjectMetadata{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: i.location,
+			Annotations: map[string]string{
+				logicalcluster.AnnotationKey: i.logicalClusterName.String(),
+			},
+		},
+	})
+	if err != nil {
+		logger.Error(err, "error building clusterKey")
+		return
+	}
+	syncTarget, err := i.syncTargetLister.Get(syncTargetKey)
```

Comment:
This will be the wrong key format - see other syncer code for the hard-coded reformat before we get new listers in.

Reply:
the original code is here https://github.com/kcp-dev/kcp/blob/main/pkg/syncer/apiimporter.go#L218-L230.

Reply:
```go
	// TODO(skuznets): can we figure out how to not leak this detail up to this code?
	// I guess once the indexer is using kcpcache.MetaClusterNamespaceKeyFunc, we can just use that formatter ...
	var indexKey string
	if upstreamNamespace != "" {
		indexKey += upstreamNamespace + "/"
	}
	if !clusterName.Empty() {
		indexKey += clusterName.String() + "|"
	}
	indexKey += name
	// get the upstream object
	obj, exists, err := c.upstreamInformers.ForResource(gvr).Informer().GetIndexer().GetByKey(indexKey)
```

From `spec_process.go` is the correct incantation.